# Workshop Nanopore Bioinformatics

A practical introduction to bioinformatics with Nanopore data on influenza for students in Namibia in the context of the Mission project.

## Schedule links for the workshop

### General
* [YYYY-MM-DD - SOMEDAY: Welcome, Linux re-cap, container, and WMS](#0)  

### Long-read bacteria genome reconstruction
* [YYYY-MM-DD - SOMEDAY: Nanopore sequencing, data formats, and QC](#1)  
* [YYYY-MM-DD - SOMEDAY: _De novo_ assembly and mapping](#2)  
* [YYYY-MM-DD - SOMEDAY: Assembly polishing and variant calling](#3)  
* [YYYY-MM-DD - SOMEDAY: Final day, Miscellaneous bacteria](#4)  

### SARS-CoV-2 data science
* [YYYY-MM-DD - SOMEDAY: FU course intro & RKI SARS-CoV-2 genomic surveillance intro](#5)  
* [YYYY-MM-DD - SOMEDAY: SARS-CoV-2 sequencing & genome reconstruction](#6)  
* [YYYY-MM-DD - SOMEDAY: SARS-CoV-2 evolution, mutation profiling & phenotypization](#7)  
* [YYYY-MM-DD - SOMEDAY: SARS-CoV-2 pathogen evolution & genome-based incidence estimation](#8)  
* [YYYY-MM-DD - SOMEDAY: SARS-CoV-2 phylogeny & outbreak investigation](#9)  

### Long-read metagenomics
* [YYYY-MM-DD - SOMEDAY: Metagenomic read classification](#10)
* [YYYY-MM-DD - SOMEDAY: Metagenome assembly, binning, qc, and annotation](#11)


## Instructors

* Franziska Kistner, Simon Tausch, Torsten Houwaart

## Schedule

> All events are held at <...>, room <...>

### <a name="0"></a> SOMEDAY, DATE
| Time        | Welcome, Linux re-cap, container & WMS |
| --          | --               |
| 10:00-10:30 | Welcome & [introduction](day-welcome-linux-container-wms/general.md) |
| 10:30-12:00 | [Linux re-cap](day-welcome-linux-container-wms/linux.md) |
| 12:00-13:00 | Lunch break |
| 13:00-14:00 | [Container & WMS](day-welcome-linux-container-wms/container-wms.md) |
| 14:00-14:30 | Coffee break |
| 14:30-16:00 | [Hands-on & demo](day-welcome-linux-container-wms/hands-on.md) |
| 16:00-16:15 | Wrap-up & questions |

### <a name="1"></a> SOMEDAY, DATE
| Time        | Nanopore |
| --          | --               |
| 09:00-10:00 | Debriefing previous day |
| 10:00-11:00 | [Nanopore intro](day-nanopore/README.md) |
| 12:00-13:00 | Lunch break |
| 13:00-13:30 | [Nanopore basecalling & data fromats](day-nanopore/README.md) |
| 13:30-14:00 | [Nanopore QC](day-nanopore/README.md) |
| 14:00-14:30 | Coffee break |
| 14:30-16:00 | [Hands-on & demo (ONT isolate)](day-nanopore/hands-on.md) |
| 14:30-16:00 | [Hands-on & demo (metagenomics)](day-nanopore/hands-on-metagenomics.md) |
| 16:00-16:15 | Wrap-up & questions |

### <a name="2"></a> SOMEDAY, DATE
| Time        | _De novo_ assembly and mapping |
| --          | --               |
| 09:00-10:00 | Debriefing previous day |
| 10:00-10:45 | [_De novo_ Assembly](day-assembly-mapping/assembly.md) |
| 10:45-11:00 | Coffee break |
| 11:00-12:00 | [Mapping & Visualization](day-assembly-mapping/mapping.md) |
| 12:00-13:00 | Lunch break |
| 13:00-14:30 | [Hands-on & demo](day-assembly-mapping/hands-on.md) |
| 14:30-15:00 | Coffee break |
| 15:00-15:45 | Continue practical session |
| 15:45-16:00 | Wrap-up & questions |

### <a name="3"></a> SOMEDAY, DATE
| Time        | Assembly polishing and variant calling |
| --          | --               |
| 09:00-10:00 | Debriefing previous day |
| 10:00-10:45 | [Assembly polishing](day-polishing-variant-calling/polishing.md) |
| 10:45-11:15 | Coffee break |
| 11:15-12:00 | [Variant calling](day-polishing-variant-calling/variant-calling.md) |
| 12:00-13:00 | Lunch break |
| 13:00-14:30 | [Hands-on & demo](day-polishing-variant-calling/hands-on.md) |
| 14:30-15:00 | Coffee break |
| 15:00-15:45 | Continue practical session |
| 15:45-16:00 | Wrap-up & questions |

### <a name="4"></a> SOMEDAY, DATE
| Time        | The Rest & Open Questions |
| --          | --               |
| 09:00-10:00 | Debriefing previous day |
| 10:00-11:00 | [Miscellaneous (bacteria)](day-misc/README.md) |
| 11:00-12:00 | [Hands-on & demo](day-misc/hands-on.md) |
| 12:00-12:30 | Wrap-up & questions |


### <a name="5"></a> SOMEDAY, DATE
| Time        | SARS-CoV-2 Introduction |
| --          | --               |
| 09:00-10:00 | Debriefing previous day |
| 10:00-11:00 | [FU course intro](day-sc2-intro/README.md) |
| 11:00-12:00 | [RKI Genomic Surveillance and SARS-CoV-2](day-sc2-intro/README.md) |

### <a name="6"></a> SOMEDAY, DATE
| Time        | SARS-CoV-2 sequencing & genome reconstruction |
| --          | --               |
| 09:00-10:00 | Debriefing previous day |
| 10:00-11:00 | [Sequencing (SARS-CoV-2)](day-sc2-seq-and-assembly/README.md) |
| 11:00-12:00 | [SARS-CoV-2 genome reconstruction](day-sc2-seq-and-assembly/README.md) |
| 12:00-13:00 | Lunch break |
| 13:00-14:30 | [Hands-on & demo](day-sc2-seq-and-assembly/hands-on.md) |
| 14:30-15:00 | Coffee break |
| 15:00-15:45 | Continue practical session |
| 15:45-16:00 | Wrap-up & questions |

### <a name="7"></a> SOMEDAY, DATE
| Time        | SARS-CoV-2 evolution, mutation profiling & phenotypization |
| --          | --               |
| 09:00-10:00 | Debriefing previous day |
| 10:00-11:00 | [SARS-CoV-2 genome organisation & evolution](day-sc2-evolution/README.md) |
| 10:00-11:00 | [SARS-CoV-2 epistasis and variants](day-sc2-evolution/README.md) |
| 12:00-13:00 | Lunch break |
| 10:00-11:00 | [SARS-CoV-2 nomenclature](day-sc2-evolution/README.md) |
| 11:00-12:00 | [SARS-CoV-2 mutation profiling](day-sc2-evolution/README.md) |
| 14:30-15:00 | Coffee break |
| 13:00-14:30 | [Hands-on & demo](day-sc2-evolution/hands-on.md) |
| 15:00-15:45 | Continue practical session |
| 15:45-16:00 | Wrap-up & questions |

### <a name="8"></a> SOMEDAY, DATE
| Time        | SARS-CoV-2 pathogen evolution & incidence estimation |
| --          | --               |
| 09:00-10:00 | Debriefing previous day |
| 10:00-12:00 | [SARS-CoV-2 pathogen evolution and genome-based incidence estimation](day-sc2-incidence/README.md) |
| 12:00-13:00 | Lunch break |
| 13:00-14:00 | [SARS-CoV-2 risk score via VOCAL](day-sc2-incidence/README.md) |
| 14:30-15:00 | Coffee break |
| 13:00-14:30 | [Hands-on & demo](day-sc2-incidence/hands-on.md) |
| 15:00-15:45 | Continue practical session |
| 15:45-16:00 | Wrap-up & questions |

### <a name="9"></a> SOMEDAY, DATE
| Time        | SARS-CoV-2 phylogeny & outbreak investigation |
| --          | --               |
| 09:00-10:00 | Debriefing previous day |
| 10:00-12:00 | [Phylogeny and outbreak investigation](day-sc2-phylo-clustering/README.md) |
| 12:00-13:00 | Lunch break |
| 13:00-14:00 | [Outbreak detection & clustering via breakfast](day-sc2-phylo-clustering/README.md) |
| 14:30-15:00 | Coffee break |
| 13:00-14:30 | [Hands-on & demo](day-sc2-phylo-clustering/hands-on.md) |
| 15:00-15:45 | Continue practical session |
| 15:45-16:00 | Wrap-up & questions |

### <a name="10"></a> SOMEDAY, DATE
| Time        | Metagenomic read classification |
| --          | --               |
| 09:00-10:00 | Debriefing previous day |
| 10:00-12:00 | [Metagenomic read classification](day-metagenomic-classification/README.md) |
| 12:00-13:00 | Lunch break |
| 13:00-14:30 | [Hands-on & demo](day-metagenomic-classification/hands-on.md) |
| 14:30-15:00 | Coffee break |
| 15:00-15:45 | Continue practical session |
| 15:45-16:00 | Wrap-up & questions |

### <a name="11"></a> SOMEDAY, DATE
| Time        | Metagenome assembly, binning, and annotation |
| --          | --               |
| 09:00-10:00 | Debriefing previous day |
| 10:00-11:00 | [Metagenome-assembled genomes (MAGs)](day-metagenomic-assembly/README.md) |
| 11:00-12:00 | [MAG quality and gene annotation](day-metagenomic-assembly/README.md) |
| 12:00-13:00 | Lunch break |
| 13:00-14:30 | [Hands-on & demo](day-metagenomic-assembly/hands-on.md) |
| 14:30-15:00 | Coffee break |
| 15:00-15:45 | Continue practical session |
| 15:45-16:00 | Wrap-up & questions |




## Acknowledgement

This course material is partly based on the following resources and on contributions from great people (no specific order):

* Martin Hoelzer, RKI MF1, content about Linux, container, Nextflow, sequencing, genomic surveillance & glueing everything together
* Sebastian "Raverjay" Krautwurst, FSU Jena, some Linux and ONT content
* Stephan Fuchs, RKI MF1, some Linux and Assembly content, Slides on SARS-CoV-2 nomenclature & phylogeny 
* Matt Huska, RKI MF1, automatic test script for all md code blocks using [codedown](https://github.com/earldouglas/codedown) and general help
* Workshop structure inspired by [https://github.com/cinemaparis/2023](https://github.com/cinemaparis/2023)
* Some ONT intro slides from Josh Quick, [original](https://github.com/cinemaparis/2023/blob/main/day1-Tuesday/slides-Quick.pdf)
* Nanopore long-read bioinformatics tutorial from [timkahlke](https://timkahlke.github.io/LongRead_tutorials)
* Max von Kleist, RKI P5 and FU Berlin, basically most content about sequencing and SC2 data science (evolution, epistasis, incidence esitmation, ...)
* Maureen Smith, Maria Trofimova, RKI P5 and FU Berlin, content on Incidence estimation
* Hugues Richard, RKI MF1, content about SC2 risk assessment
* Matt Huska & Denis Beslic, RKI MF1, content about SC2 outbreak detection & clustering
